File: DSA Using Java/Pattern Printing/ButterflyPattern.java
import java.util.Scanner;

public class ButterflyPattern {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number : ");
        int n = sc.nextInt();

        // First Half
        for (int i = 1; i <= n; i++){
            // stars - i
            for(int j = 1; j <= i; j++){
                System.out.print("* ");
            }

            // spaces - 2*(n-i)
            for(int j = 1; j <= 2*(n-i); j++){
                System.out.print("  ");
            }

            // stars - i
            for(int j = 1; j <= i; j++){
                System.out.print("* ");
            }

            // Next line
            System.out.println();
        }

        // Second Half
        for (int i = n; i >= 1; i--){
            // stars - i
            for(int j = 1; j <= i; j++){
                System.out.print("* ");
            }

            // spaces - 2*(n-i)
            for(int j = 1; j <= 2*(n-i); j++){
                System.out.print("  ");
            }

            // stars - i
            for(int j = 1; j <= i; j++){
                System.out.print("* ");
            }

            // Next line
            System.out.println();
        }
    }
}

File: DSA Using C++/PatternPrinting/butterfly-pattern.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cout << "Enter a number: ";
    cin >> n;

    // first part
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j <= i; j++)
        {
            cout << "* ";
        }

        for (int j = 0; j < 2 * (n - i - 1); j++)
        {
            cout << "  ";
        }

        for (int j = 0; j <= i; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }

    // second part
    for (int i = n - 1; i >= 0; i--)
    {
        for (int j = 0; j <= i; j++)
        {
            cout << "* ";
        }

        for (int j = 0; j < 2 * (n - i - 1); j++)
        {
            cout << "  ";
        }

        for (int j = 0; j <= i; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }

    return 0;
}